pkg/controller/migplan/migplan_controller.go

```diff
@@ -157,9 +158,15 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 var _ reconcile.Reconciler = &ReconcileMigPlan{}
 
+type Identities struct {
+	SrcIdentity  *migauth.Identity
+	DestIdentity *migauth.Identity
+}
+
 // ReconcileMigPlan reconciles a MigPlan object
 type ReconcileMigPlan struct {
 	client.Client
+	Identities
```

Comment:
This is a interesting convenience but a little concerned.  The pattern so far has been to pass around the reconciled resource (or related information).  I wonder if we can achieve the same convenience by adding the identities to the `PlanResources`.

That said, perhaps this is a good _new_ president.  Now that a lot of code has been written for the Reconcilers stashing the plan (and related information) may be better than passing it around everywhere.  Wonder why we never did that?

In either case, I'm not convinced the `Identities` aggregation provides much value.  Just my $0.02.
```
type ReconcileMigPlan struct {
	client.Client
	scheme *runtime.Scheme
        plan *migapi.MigPlan
        srcIdentity *migauth.Identity
        dstIdentity *migauth.Identity
```

The `plan` would be `set` at the beginning of the reconcile.  The identities could be set after validate().  both in the main (though identities probably in a separate method  like you have it outside the loop).
Sure would clean up a lot of method signatures.

What do you think?

Reply:
Have a couple questions about this I will sync up with you in the morning.